docs/source/amp.rst

```diff
@@ -0,0 +1,365 @@
+.. role:: hidden
+    :class: hidden-section
+
+Automatic Mixed Precision package - torch.amp
+==================================================
+
+.. automodule:: torch.amp
+.. currentmodule:: torch.amp
+
+``torch.amp`` provides convenience methods for mixed precision:  using fast FP16 arithmetic where you can and
+stable FP32 arithmetic where you must to optimize speed while maintaining accuracy and stability.
+
+The two ingredients of the mixed precision recipe are `Autocasting`_ and `Gradient Scaling`_.
+
+By default, you don't need to call ``.half()`` on your model(s) or data to use the routines below.  In fact, you shouldn't:
+model params should remain FP32.
+
+Turning on mixed precision should not require retuning any hyperparameters, as long as the conventions shown in the `Examples`_
+below are obeyed.
+
+.. contents:: :local:
+
+Autocasting
+^^^^^^^^^^^
+
+Under construction...
+
+Gradient scaling
```

Comment:
This jumps in too deep with a lot of assumed context to a particular type of auto-precision. Can we simplify this document by focusing on the exposed methods, what they do, and then suggesting when they could be useful?